pex/layout.py

```diff
@@ -224,14 +460,42 @@ def dist_strip_prefix(self, dist_name):
         # type: (str) -> Optional[str]
         return os.path.join(DEPS_DIR, dist_name)
 
+    def dist_size(
+        self,
+        dist_relpath,  # type: str
+        is_wheel_file,  # type: bool
+    ):
+        # type: (...) -> int
+        if is_wheel_file:
+            return self._zfp.getinfo(dist_relpath).file_size
+        else:
+            return sum(
+                self._zfp.getinfo(name).file_size
+                for name in self._names
+                if name.startswith(dist_relpath)
```

Comment:
I know this pattern is used elsewhere, but just asking to confirm that one can't have two unrelated dists where the `dist_relpath` of one will be a string prefix of the other? (e.g. `a/b` vs. `a/bc`)

For instance, does `dist_relpath` always end in `/`?

Reply:
A dist relative path is `.deps/*.whl`. Using the wheel name here is critical to PEX functioning (runtime dist resolution perf in multi-platform PEXes). Since the suffix is fixed and conspires to be illegal in any of the 5 possible wheel name component slots, this is safe.